core/src/main/java/org/apache/iceberg/rest/requests/UpdateNamespacePropertiesRequest.java

```diff
@@ -0,0 +1,112 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.requests;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+
+/**
+ * A REST request to set and/or remove properties on a namespace.
+ * <p/>
+ * The request body specifies a list of properties to remove and a map
+ * of key value pairs to update.
+ * <p/>
+ * Properties that are not in the request are not modified or removed by this call.
+ * Server implementations are not required to support namespace properties.
+ */
+public class UpdateNamespacePropertiesRequest {
+
+  private List<String> removals;
+  private Map<String, String> updates;
+
+  public UpdateNamespacePropertiesRequest() {
+    // Required for Jackson
+  }
+
+  private UpdateNamespacePropertiesRequest(List<String> removals, Map<String, String> updates) {
+    this.removals = removals;
+    this.updates = updates;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public List<String> removals() {
+    return ImmutableList.copyOf(removals);
+  }
+
+  public Map<String, String> updates() {
+    return ImmutableMap.copyOf(updates);
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("removals", removals())
+        .add("updates", updates())
+        .toString();
+  }
+
+  public static class Builder {
+    private final ImmutableSet.Builder<String> removalsBuilder;
+    private final ImmutableMap.Builder<String, String> updatesBuilder;
+
+    private Builder() {
+      this.removalsBuilder = ImmutableSet.builder();
+      this.updatesBuilder = ImmutableMap.builder();
+    }
+
+    public Builder withRemoval(String removal) {
+      removalsBuilder.add(removal);
+      return this;
+    }
+
+    public Builder withRemovals(Iterable<String> removals) {
+      removalsBuilder.addAll(removals);
+      return this;
+    }
+
+    public Builder withUpdate(String key, String value) {
+      updatesBuilder.put(key, value);
+      return this;
+    }
+
+    public Builder withUpdates(Map<String, String> updates) {
+      updatesBuilder.putAll(updates);
+      return this;
+    }
+
+    public UpdateNamespacePropertiesRequest build() {
+      List<String> removals = removalsBuilder.build().asList();
+      Map<String, String> updates = updatesBuilder.build();
+      Preconditions.checkState(
+          !removals.isEmpty() || !updates.isEmpty(),
+          "Cannot create a request to update a namespace's properties without any properties to remove or update.");
```

Comment:
I'd probably remove these. The requests are likely to be built and automated. Better to have an `isEmpty` method on the request, but I probably wouldn't do that either.